test/stability.js

```diff
@@ -88,3 +89,52 @@ describe("TestRPC", function(done) {
     batch.execute();
   });
 });
+
+describe("race conditions", function(done) {
+  var web3 = new Web3();
+  var provider;
+  var accounts;
+
+  before("initialize the provider", function() {
+    provider = TestRPC.provider();
+    web3.setProvider(provider);
+  });
+
+  before("get accounts", function(done) {
+    web3.eth.getAccounts(function(err, accs) {
+      if (err) return done(err);
+
+      accounts = accs;
+      done();
+    });
+  });
+
+  it("should not cause 'get' of undefined", function(done) {
+    process.prependOnceListener('uncaughtException', function(err) {
+      done(err);
+    });
+
+    var blockchain = provider.manager.state.blockchain;
+    blockchain.vm.stateManager.checkpoint(); // processCall or processBlock
+    blockchain.stateTrie.get(utils.toBuffer(accounts[0]), function() {}); // getCode (or any function that calls trie.get)
+    blockchain.vm.stateManager.revert(function() {}); // processCall or processBlock
+  });
+
+  it("should not cause 'pop' of undefined", function(done) {
+    process.prependOnceListener('uncaughtException', function(err) {
+      done(err);
+    });
+
+    var blockchain = provider.manager.state.blockchain;
+    blockchain.vm.stateManager.checkpoint(); // processCall #1
+    // processNextBlock triggered by interval mining which at some point calls vm.stateManager.commit() and blockchain.putBlock()
+    blockchain.processNextBlock(function(err, tx, results) {
+      blockchain.vm.stateManager.revert(function() { // processCall #1 finishes
+        blockchain.latestBlock(function (err, latestBlock) { 
+          blockchain.stateTrie.root = latestBlock.header.stateRoot; // getCode #1 (or any function with this logic)
```

Comment:
This might not be possible because the function calls cleans up after itself, ie. 
```
// Finally, put the stateRoot back for good
trie.root = currentStateRoot;
```

The scenario requires certain execution to be intertwined

I did a quick test with getAccount and the test passed when it should have failed.